### ISSUE TITLE
Use the new implementation of MpiVariableSynchronizeDispatcher by default

### ISSUE THREAD

**Issue by hmt23** [arcaneframework/framework#185]
I replace the code of the MpiVariableSynchronizeDispatcher to reimplement the synchronize function when MPI is used. I think this code is better than the old one for readability and maybe for performance. I go faster on 4 cores but we have to check on real apps.

Please check the code before.

**Comment by hmt23** [comment]
#184

**Comment by grospelliergilles** [comment]
It may be faster but before making it the default we need intensive tests on all production machines (with large core count) and compare the performances with the current version.

**Comment by hmt23** [comment]
I totally agree with you about testing intensively on all our productions machines before.
It is just to remind me to make all these tests and discuss when the code in #184 can or cannot be the default one.
Furthermore, it is the first version, and the aim is to simplify the code by removing all the mpi request arrays. I will use MPI directly in order to improve performance and robustness.

**Comment by grospelliergilles** [comment]
This implementation is not faster than the current default one son we will not change the default.
